packages/cip30/src/Wallet/Wallet.ts

```diff
@@ -24,67 +31,88 @@ export type RequestAccess = () => Promise<boolean>;
 
 export type WalletOptions = {
   logger?: Logger;
-  persistAllowList?: boolean;
-  storage?: Storage;
+  storage?: Storage.LocalStorageArea;
 };
 
 const defaultOptions = {
   logger: dummyLogger,
   persistAllowList: false,
-  storage: window.localStorage
+  storage: storage.local
+};
+
+type WalletStorage = {
+  allowList: string[];
 };
 
 export class Wallet {
-  readonly version: SpecificationVersion;
+  readonly apiVersion: ApiVersion;
   readonly name: WalletName;
+  readonly icon: WalletIcon;
 
   private allowList: string[];
   private logger: Logger;
   private readonly options: Required<WalletOptions>;
 
-  constructor(
+  private constructor(
     properties: WalletProperties,
     private api: WalletApi,
     private requestAccess: RequestAccess,
+    allowList: string[],
     options?: WalletOptions
   ) {
     this.options = { ...defaultOptions, ...options };
     this.name = properties.name;
-    this.version = properties.version;
-    this.allowList = this.options.persistAllowList ? this.getAllowList() : [];
+    this.apiVersion = properties.apiVersion;
+    this.icon = properties.icon;
+    this.allowList = allowList;
     this.logger = this.options.logger;
   }
 
-  public getPublicApi(window: WindowMaybeWithCardano) {
+  public getPublicApi(hostname: string) {
     return {
-      enable: this.enable.bind(this, window),
-      isEnabled: this.isEnabled.bind(this, window),
-      name: this.name,
-      version: this.version
+      apiVersion: this.apiVersion,
+      enable: this.enable.bind(this, hostname),
+      icon: this.icon,
+      isEnabled: this.isEnabled.bind(this, hostname),
+      name: this.name
     };
   }
 
-  private getAllowList(): string[] {
-    // JSON.parse(null) seems to be legit
-    return JSON.parse(this.options.storage.getItem(this.name)!) || [];
+  static async getAllowList(_storage?: Storage.LocalStorageArea): Promise<string[]> {
+    if (!_storage) return Promise.resolve([]);
+    try {
+      const persistedStorage: Record<string, WalletStorage> = await _storage.get(this.name);
```

Comment:
`this.name` inside the static function refers to function name (not the name property declared by this class) which is probably not what we want here.